tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,157 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"testing"
+	"time"
+)
+
+// TODO: A neater way to provide ocdev path. Currently we assume \
+// ocdev and oc in $PATH already.
+
+func runCmd(cmdS string) string {
+	cmd, err := exec.Command("/bin/sh", "-c", cmdS).Output()
+	if err != nil {
+		log.Fatalf("Error running command %s: %s", cmdS, err)
+		os.Exit(1)
+	}
+	return string(cmd)
+}
+
+func createApp(appName string) {
+	runCmd("ocdev application create " + appName)
+}
+
+func getApp() string {
+	app := runCmd("ocdev application get -q")
+	return app
+}
+
+func getCmp() string {
+	cmp := runCmd("ocdev component get --short")
+	return cmp
+}
+
+func pingSvc(url string) {
+	var ep int
+	pingTimeout := time.After(10 * time.Minute)
+	tick := time.Tick(time.Second)
+
+	for {
+		select {
+		case <-pingTimeout:
+			fmt.Errorf("could not ping the specific service in given time: 10 minutes")
+			os.Exit(1)
+
+		case <-tick:
+			httpTimeout := time.Duration(10 * time.Second)
+			client := http.Client{
+				Timeout: httpTimeout,
+			}
+
+			response, err := client.Get(url)
+			if err != nil {
+				time.Sleep(1 * time.Second)
+				continue
+			}
+
+			if response.Status == "200 OK" {
+				ep = 1
+
+			} else {
+				fmt.Errorf("for service")
+				os.Exit(1)
+			}
+		}
+
+		if ep > 0 {
+			break
+		}
+	}
+}
+
+func TestOCdev(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "ocdev test suite")
+}
+
+var _ = BeforeSuite(func() {
+	runCmd("oc cluster down")
+	runCmd("oc cluster up")
```

Comment:
@kadel need you input on this: do you think we should deal with `oc cluster up/down` outside of e2e scripts. One advantage with that would be to allow running tests with minishift.

Reply:
It might be a good idea. It will also allow easily switch between OpenShift versions